superset/thumbnails/digest.py

```diff
@@ -49,8 +52,40 @@ def _adjust_string_for_executor(
     return unique_string
 
 
+def _adjust_string_with_rls(
+    unique_string: str,
+    datasources: list[SqlaTable | None] | set[BaseDatasource],
+    executor: str,
+) -> str:
+    """
+    Add the RLS filters to the unique string based on current executor.
+    """
+    user = (
+        security_manager.find_user(executor)
+        or security_manager.get_current_guest_user_if_guest()
+    )
+
+    if user:
+        stringified_rls = ""
+        with override_user(user):
+            for datasource in datasources:
+                if (
+                    datasource
+                    and hasattr(datasource, "is_rls_supported")
+                    and datasource.is_rls_supported
+                ):
+                    rls_filters = datasource.get_sqla_row_level_filters()
+                    stringified_rls += "".join([str(f) for f in rls_filters])
```

Comment:
I think it may be a good idea to append the dataset id to the RLS string, just to make sure we don't introduce potential leaks. Something like `<id>\t<rls_filters>`. Also, I think it may be a good idea to use some other separator than an empty string(`\t` is probably a safe bet) and also finish it off with a `\n`, also for the same reason.